src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -66,21 +126,59 @@ else if (stateSpaceType.equals("Inertia"))
                 NOMINAL_VOLTAGE, LOOP_PERIOD);
     }
 
+    /**
+     * Gets the velocity of the motor.
+     *
+     * @return the velocity of the motor. [rpm]
+     */
     public double getVelocity() {
-        return unitModel.toUnits(mainMotor.getSelectedSensorVelocity());
+        if (Robot.isSimulation()) {
+            return flywheelSim.getAngularVelocityRPM();
+        }
+        return unitModel.toVelocity(motor.getSelectedSensorVelocity());
     }
 
-    public void setVelocity(double velocity, double timeInterval) {
-        velocity = Utils.deadband(velocity, WPI_TalonFX.kDefaultSafetyExpiration);
-
-        linearSystemLoop.setNextR(VecBuilder.fill(velocity));
-        linearSystemLoop.correct(VecBuilder.fill(getVelocity()));
-        linearSystemLoop.predict(timeInterval);
+    /**
+     * Sets the velocity of the motor.
+     *
+     * @param velocity is the velocity setpoint. [rpm]
+     */
+    public void setVelocity(double velocity) {
+        linearSystemLoop.setNextR(VecBuilder.fill(Units.rotationsToRadians(Utils.secondsToMinutes(velocity))));
```

Comment:
the inputted velocity is in rpm, so:
the velocity is converted from seconds to minutes (same as minutes to hours): the units are [rph] -> rotations to radians: units: radians per hour...

Reply:
Once again: The units are not in minutes, they are in rotation/minute. So if we substitute minute = second * 60, all in all we divide by 60. Hence seconds to minutes. Should I rename the function to rpmToRps?

Reply:
You should definitely rename the function.